--- C:/tools/msys64/opt/devkitpro/devkitA64/base_tools	2020-04-18 03:05:23 +0000
+++ C:/tools/msys64/opt/devkitpro/devkitA64/base_tools_	2020-04-18 02:53:34 +0000
@@ -25,14 +25,14 @@ export PORTLIBS_PATH	:=	$(DEVKITPATH)/po
 #---------------------------------------------------------------------------------
 PREFIX		:=	aarch64-none-elf-
 
-export CC	:=	$(PREFIX)gcc
-export CXX	:=	$(PREFIX)g++
-export AS	:=	$(PREFIX)as
-export AR	:=	$(PREFIX)gcc-ar
-export OBJCOPY	:=	$(PREFIX)objcopy
-export STRIP	:=	$(PREFIX)strip
-export NM	:=	$(PREFIX)gcc-nm
-export RANLIB	:=	$(PREFIX)gcc-ranlib
+override CC	:=	$(PREFIX)gcc
+override CXX	:=	$(PREFIX)g++
+override AS	:=	$(PREFIX)as
+override AR	:=	$(PREFIX)gcc-ar
+override OBJCOPY	:=	$(PREFIX)objcopy
+override STRIP	:=	$(PREFIX)strip
+override NM	:=	$(PREFIX)gcc-nm
+override RANLIB	:=	$(PREFIX)gcc-ranlib
 
 ISVC=$(or $(VCBUILDHELPER_COMMAND),$(MSBUILDEXTENSIONSPATH32),$(MSBUILDEXTENSIONSPATH))
 
