diff -uprN libretro-super/libretro-build-libnx.sh libretro-super_/libretro-build-libnx.sh
--- libretro-super/libretro-build-libnx.sh	1970-01-01 09:00:00.000000000 +0900
+++ libretro-super_/libretro-build-libnx.sh	2019-03-12 12:55:41.856089000 +0900
@@ -0,0 +1,17 @@
+#! /usr/bin/env bash
+# vim: set ts=3 sw=3 noet ft=sh : bash
+
+SCRIPT="${0#./}"
+BASE_DIR="${SCRIPT%/*}"
+WORKDIR="$PWD"
+
+if [ "$BASE_DIR" = "$SCRIPT" ]; then
+	BASE_DIR="$WORKDIR"
+else
+	if [[ "$0" != /* ]]; then
+		# Make the path absolute
+		BASE_DIR="$WORKDIR/$BASE_DIR"
+	fi
+fi
+
+platform=libnx ${BASE_DIR}/libretro-build.sh $@

