diff -uprN libretro-super/libretro-build-libnx.sh libretro-super_/libretro-build-libnx.sh
--- libretro-super/libretro-build-libnx.sh	1970-01-01 09:00:00.000000000 +0900
+++ libretro-super_/libretro-build-libnx.sh	2019-03-20 06:35:21.600298900 +0900
@@ -0,0 +1,17 @@
+#! /usr/bin/env bash
+# vim: set ts=3 sw=3 noet ft=sh : bash
+
+SCRIPT="${0#./}"
+BASE_DIR="${SCRIPT%/*}"
+WORKDIR="$PWD"
+
+if [ "$BASE_DIR" = "$SCRIPT" ]; then
+	BASE_DIR="$WORKDIR"
+else
+	if [[ "$0" != /* ]]; then
+		# Make the path absolute
+		BASE_DIR="$WORKDIR/$BASE_DIR"
+	fi
+fi
+
+platform=libnx ${BASE_DIR}/libretro-build.sh $@
diff -uprN libretro-super/libretro-config.sh libretro-super_/libretro-config.sh
--- libretro-super/libretro-config.sh	2019-03-20 06:51:19.098789600 +0900
+++ libretro-super_/libretro-config.sh	2019-03-20 06:54:51.895701300 +0900
@@ -600,6 +600,9 @@ case "$platform" in
 		FORMAT_COMPILER_TARGET_ALT=libnx
 		FORMAT=_libnx
 
+		CC="$DEVKITA64/bin/aarch64-none-elf-gcc$BINARY_EXT"
+		CXX="$DEVKITA64/bin/aarch64-none-elf-g++$BINARY_EXT"
+		AR="$DEVKITA64/bin/aarch64-none-elf-ar$BINARY_EXT"
 		;;
 
 	sncps3)
@@ -851,6 +854,12 @@ case "$platform" in
 					FORMAT_COMPILER_TARGET="ngc"
 					DIST_DIR="ngc"
 					;;
+				*libnx*)
+					platform=libnx
+					FORMAT_EXT="a"
+					FORMAT_COMPILER_TARGET="libnx"
+					DIST_DIR="libnx"
+					;;
 				android-x86_64)
 					platform=android-x86_64
 					FORMAT_EXT="so"

